src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -3030,5 +3030,125 @@ public object Y
+                // (8,25): error CS1525: Invalid expression term ')'
+                //         if (o is A[]? b5) { }         // error 3 (missing :)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(8, 25),
+                // (9,23): error CS0103: The name 'b6' does not exist in the current context
+                //         if (o is A[]? b6 && c) { }    // error 4 (missing :)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "b6").WithArguments("b6").WithLocation(9, 23),
+                // (9,30): error CS1003: Syntax error, ':' expected
+                //         if (o is A[]? b6 && c) { }    // error 4 (missing :)
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(":", ")").WithLocation(9, 30),
+                // (9,30): error CS1525: Invalid expression term ')'
+                //         if (o is A[]? b6 && c) { }    // error 4 (missing :)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(9, 30),
+                // (10,25): error CS0103: The name 'b7' does not exist in the current context
+                //         if (o is A[][]? b7) { }       // error 5 (missing :)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "b7").WithArguments("b7").WithLocation(10, 25),
+                // (10,27): error CS1003: Syntax error, ':' expected
+                //         if (o is A[][]? b7) { }       // error 5 (missing :)
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(":", ")").WithLocation(10, 27),
+                // (10,27): error CS1525: Invalid expression term ')'
+                //         if (o is A[][]? b7) { }       // error 5 (missing :)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(10, 27),
+                // (11,25): error CS0103: The name 'b8' does not exist in the current context
+                //         if (o is A[][]? b8 && c) { }  // error 6 (missing :)
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "b8").WithArguments("b8").WithLocation(11, 25),
+                // (11,32): error CS1003: Syntax error, ':' expected
+                //         if (o is A[][]? b8 && c) { }  // error 6 (missing :)
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(":", ")").WithLocation(11, 32),
+                // (11,32): error CS1525: Invalid expression term ')'
+                //         if (o is A[][]? b8 && c) { }  // error 6 (missing :)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, ")").WithArguments(")").WithLocation(11, 32)
+                );
+        }
+
+        [Fact]
+        public void NullableArrayDeclarationPattern_Good_01()
+        {
+            var source =
+@"#nullable enable
+public class A
+{
+    static void M(object o, bool c)
+    {
+        if (o is A?) { }                 // ok 1 (for compat)
```

Comment:
What compat?shouldn't this be an error per https://github.com/dotnet/roslyn/issues/32025?